src/main/java/com/denizenscript/depenizen/bukkit/bridges/LuckPermsBridge.java

```diff
@@ -1,66 +1,75 @@
-                tracks.addObject(new LuckPermsTrackTag(track));
-            }
-            event.setReplacedObject(tracks.getObjectAttribute(attribute.fulfill(1)));
+        public LuckPermsBridgeTags() {
+            instance = this;
+            TagManager.registerStaticTagBaseHandler(LuckPermsBridgeTags.class, "luckperms", (t) -> instance);
         }
 
-        // <--[tag]
-        // @attribute <luckperms.track[<track_name>]>
-        // @returns LuckPermsTrackTag
-        // @plugin Depenizen, LuckPerms
-        // @description
-        // Returns the track from the name given.
-        // -->
-        if (attribute.startsWith("track")) {
-            if (attribute.hasParam()) {
-                event.setReplacedObject(attribute.paramAsType(LuckPermsTrackTag.class).getObjectAttribute(attribute.fulfill(1)));
-            }
+        public void register() {
+
+            // <--[tag]
+            // @attribute <luckperms.list_tracks>
+            // @returns ListTag(LuckPermsTrackTag)
+            // @plugin Depenizen, LuckPerms
+            // @description
+            // Returns a list of all tracks.
+            // -->
+            tagProcessor.registerTag(ListTag.class, "list_tracks", (attribute, object) -> {
+                return new ListTag(luckPermsInstance.getTrackManager().getLoadedTracks(), LuckPermsTrackTag::new);
+            });
+
+            // <--[tag]
+            // @attribute <luckperms.track[<track_name>]>
+            // @returns LuckPermsTrackTag
+            // @plugin Depenizen, LuckPerms
+            // @description
+            // Returns the track from the name given.
+            // -->
+            tagProcessor.registerTag(LuckPermsTrackTag.class, ElementTag.class, "track", (attribute, object, name) -> {
+                Track track = luckPermsInstance.getTrackManager().getTrack(name.toString());
+                return track != null ? new LuckPermsTrackTag(track) : null;
+            });
+
+            // <--[tag]
+            // @attribute <luckperms.list_groups>
+            // @returns ListTag(LuckPermsGrupTag)
+            // @plugin Depenizen, LuckPerms
+            // @description
+            // Returns a list of all luckperms groups.
+            // -->
+            tagProcessor.registerTag(ListTag.class, "list_groups", (attribute, object) -> {
+                return new ListTag(luckPermsInstance.getGroupManager().getLoadedGroups(), LuckPermsGroupTag::new);
+            });
         }
+
+    }
+
+    public static LuckPerms luckPermsInstance;
+
+    @Override
+    public void init() {
+        luckPermsInstance = LuckPermsProvider.get();
+        LuckPermsPlayerExtensions.register();
+        ObjectFetcher.registerWithObjectFetcher(LuckPermsGroupTag.class, LuckPermsGroupTag.tagProcessor).generateBaseTag();
+        ObjectFetcher.registerWithObjectFetcher(LuckPermsTrackTag.class, LuckPermsTrackTag.tagProcessor).generateBaseTag();
```

Comment:
`generateBaseTag` is what makes an object's constructor tag - should either add proper meta or remove that (in the case of groups you probably want it since there's no alternative, in case of tracks there's `luckperms.track[<name>]` already, so if you want to add a proper constructor for it you'll need to deprecate that tag).

Although that automatically uses the object's name, which will make something like `<luckpermsgroup[<name>]>` - might be nicer to register it manually and make it `luckperms_group` or something like that.